clients/da-vinci-client/src/main/java/com/linkedin/davinci/VersionBackend.java

```diff
@@ -95,8 +101,15 @@ public class VersionBackend {
     this.stopConsumptionTimeoutInSeconds =
         backend.getConfigLoader().getCombinedProperties().getInt(SERVER_STOP_CONSUMPTION_TIMEOUT_IN_SECONDS, 60);
     this.storeDeserializerCache = backend.getStoreOrThrow(store.getName()).getStoreDeserializerCache();
-    this.compressor = Lazy.of(
-        () -> backend.getCompressorFactory().getCompressor(version.getCompressionStrategy(), version.kafkaTopicName()));
+
+    // If recordTransformer is enabled, we need to disable compression since it is already done by the recordTransformer
+    if (recordTransformer != null) {
```

Comment:
Why is this necessary?  Does this mean that decompression was being done earlier?

Reply:
Oh wait, actually I suppose this makes sense.  Davinci would be storing the data compressed and decompressing on read I think.  Do I have it right?

Reply:
I faced an issue in an earlier commit where the data already was decompressed at an earlier stage with record transformation on, but I can't replicate it anymore. Let me push a change without this to ensure that it passes every single CI stage.

Reply:
It's possible that when the data was written the version was still in NoOp and by the time it tried to read compression was set to GZIP.